### PR TITLE
Add team and player models and controllers to view collections

### PR DESCRIPTION
**Pull request by AEgan**
This pull request adds the team model, controller, and views for show and index actions, as well as a player model, controller, and a show action.

Currently this change allows users to view team collections. They can browse to /teams or click the link to 'Collections' in navigation bar, and see all of the teams as well as the number of players needed to complete that team's collection. Viewing an individual teams page will show all of the players on the team, as well as that player's position, playing style, and overall in-game rating. This information is important to users when looking up the players on the in-game market, allowing them to narrow down their search, as NHL 15 does not support searching by player names.

This addition will also allow us to create trade offers and collateral in future updates to the site, as well as allow users to search for people who are attempting to trade, borrow, or loan individual players.

There is a doc folder that includes a csv file 'teams.csv' which includes the city and name of every team in the NHL. There is also a teams/ folder that includes files with all of the players for each team. The individual file names the names of the teams, in all lower case, with an underscore instead of spaces. These files are used to populate the database.

@cmar @ashirazi

### CODE REVIEW

**Comment by AEgan** [comment]
This is deployed to heroku at sleepy-bayou-2110.herokuapp.com

to sign in, you can use the username 'egan' and password 'secret'

Nothing about the functionality in this PR is affected by someone logging in, but if you want to sign in or create an account feel free!

**Comment by AEgan** [comment]
I added  `dependent: :destroy` to my team's `has_many` relationship, and improved the `player_belongs_to_team` validation. 

Thanks for the feedback @ashirazi !

File: app/controllers/teams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class TeamsController < ApplicationController
+  def index
+    @teams = Team.alphabetical.includes(:players)
```
Comment (cmar) [inline app/controllers/teams_controller.rb:3]: nice use of scope!

glad to see your thinking ahead and `includes(:players)`

File: db/migrate/20150723193309_create_players.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def change
+    create_table :players do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :position
+      t.integer :team_id
+      t.integer :overall
+      t.string :style
+
+      t.timestamps null: false
```
Comment (cmar) [inline db/migrate/20150723193309_create_players.rb:11]: Would Players benefit from an index on `team_id`?

File: db/schema.rb [hunk 3]
```diff
@@ -11,11 +11,29 @@
...
+  create_table "teams", force: :cascade do |t|
+    t.string   "name"
+    t.string   "city"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
   create_table "users", force: :cascade do |t|
     t.string   "username"
     t.string   "password_digest"
```
Comment (cmar) [inline db/schema.rb]: try `config.active_record.schema_format = :sql` to get some experience with db/structure.sql files

**Comment by cmar** [comment]
@AEgan this is very clean. Nice job!

**Comment by nhessler** [comment]
very nice! :+1:

**Comment by AEgan** [comment]
Thanks for the feedback! I've changed the schema_format to get an idea of what's happening, and added an index on team_id for the players table.

**Pull request merged**